cwms-data-api/src/main/java/cwms/cda/data/dao/LocationGroupDao.java

```diff
@@ -216,6 +216,95 @@ public List<LocationGroup> getLocationGroups(@Nullable String officeId,
         }
     }
 
+    /**
+     * Get all location groups for a given office and category,
+     * as well as a where clause to filter the shared_ref_location_id.
+     * @Param officeId The office id to use for the query.
+     * @Param locCategoryLike A regex to use to filter the location categories.  May be null.
+     * @Param sharedRefLocLike A where clause to filter the shared_loc_alias_id.  May be null.
+     * @return A list of all location groups for the given parameters.
+     */
+
+    public List<LocationGroup> getLocationGroups(String officeId, String locCategoryLike, String sharedRefLocLike) {
```

Comment:
Unit test this since it's part of the public LocationGroupDao API

Reply:
Added unit test for new functionality